File: Data Structures & Algorithms/first-missing-positive/submission-2.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        vector<int> cnt(100000,0);
        int mn = *min_element(nums.begin(), nums.end());
        if(mn>1)    return 1;
        for(auto ele: nums){
            cnt[ele-mn]++;
        }
        int x = 0;
        for(int i=0; i<cnt.size(); i++){
            if(cnt[i]==0 && i+mn>0)   return i+mn;
        }
        return 100001;
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        mp = {}
        s = set()
        i,j = 0,0
        while i < n:
            if i-j > k:
                s.remove(nums[j])
                j += 1
            if nums[i] in s:
                return True
            s.add(nums[i])
            i += 1
        return False